LongWarOfTheChosen/Src/LW_Overhaul/Classes/UIScreenListener_StrategyMap_HavenManagementButton.uc

```diff
@@ -0,0 +1,88 @@
+//---------------------------------------------------------------------------------------
+//  AUTHOR:  Rai 
+//  PURPOSE: This class adds the corresponding screen listener to access the haven management 
+//           screen from the geoscape directly
+//---------------------------------------------------------------------------------------
+//  Credits: Adapted from WOTCStrategyOverhaul Team's code for CI
+//---------------------------------------------------------------------------------------
+
+class UIScreenListener_StrategyMap_HavenManagementButton extends UIScreenListener;
+
+/////////////////////
+/// Adding button ///
+/////////////////////
+
+event OnInit(UIScreen Screen)
+{
+	local UIStrategyMap_HavenManagementButton HavenManagementButton;
+	local UIStrategyMap StrategyMap;
+
+	StrategyMap = UIStrategyMap(Screen);
+	if (StrategyMap == none) return;
+
+	HavenManagementButton = Screen.Spawn(class 'UIStrategyMap_HavenManagementButton', StrategyMap.StrategyMapHUD);
```

Comment:
This could really do with a tooltip, just like the faction HQ icons have (and the black market, etc.)

Reply:
Adding a tooltip is apparently real complicated, as is also pointed out by "@sbatista" in the code for UIPanel. 
To get around that and also communicate to the player that the icon being added is clickable, I thought I added a resizing/recoloring effect to the text itself, making it look like it is animated on mouseover. It seems that was missed in the commit, so I am adding that now in "UIStrategyMap_HavenManagementButton.uc"